src/backend/distributed/executor/multi_utility.c

```diff
@@ -336,6 +336,17 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 		ErrorIfUnsupportedTruncateStmt((TruncateStmt *) parsetree);
 	}
 
+	/* prevent any operation on pg_dist_node while dropping the extension */
```

Comment:
This requires a bit more elaboration: Deadlock detection starts by taking an `AccessShareLock` on `pg_dist_node` followed by `pg_dist_local_group`, but `DROP EXTENSION` may take `AccessExclusiveLock` on those table in a different order, which could lead to unexpected deadlocks.